### PR TITLE
Moving ls_r function (analogue of `ls -r`) to dirmanager

### PR DESCRIPTION
**Pull request by inexxt**
This PR adds `list_dir_recursive` function to dirmanager. The function is an analogue to `ls -r` from Linux.
It was previously defined in `dummytaskstate.py`, but was intended to be moved to dirmanager - it was done in https://github.com/golemfactory/golem/pull/1297 but since that PR is probably never going to be merged, I'm adding it separately here.

### CODE REVIEW

File: golem/resource/dirmanager.py [hunk 1]
```diff
@@ -55,6 +55,14 @@ def find_task_script(task_dir, script_name):
     logger.error("Script file {} does not exist!".format(script_file))
 
 
+def ls_r(dir):
```
Comment (Wiezzel) [inline golem/resource/dirmanager.py]: This is a really bad function name. I'd suggest something more informative, e.g. `list_dir_recursive`.

File: golem/resource/dirmanager.py [hunk 2]
```diff
@@ -55,6 +55,14 @@ def find_task_script(task_dir, script_name):
     logger.error("Script file {} does not exist!".format(script_file))
 
 
+def ls_r(dir):
+    files = []
+    for dirpath, dirnames, filenames in os.walk(dir, followlinks=True):
+        for name in filenames:
+            files.append(os.path.join(dirpath, name))
```
Comment (Wiezzel) [inline golem/resource/dirmanager.py]: Maybe `yield` instead? Caller could always wrap the call in `list()` if the result needs to be reused.

File: golem/resource/dirmanager.py (hunk 1 above)
Comment (Wiezzel) [reply golem/resource/dirmanager.py]: Type annotation would also be nice.

File: golem/resource/dirmanager.py [hunk 3]
```diff
@@ -55,6 +56,12 @@ def find_task_script(task_dir, script_name):
     logger.error("Script file {} does not exist!".format(script_file))
 
 
+def list_dir_recursive(dir: str) -> Generator[str]:
```
Comment (Wiezzel) [inline golem/resource/dirmanager.py]: `Generator` takes 3 params. The return type should be `Iterator[str]`.

File: tests/apps/dummy/task/test_dummytaskstate.py [hunk 4]
```diff
@@ -3,8 +3,9 @@
 from unittest.mock import patch
 
 from apps.dummy.dummyenvironment import DummyTaskEnvironment
-from apps.dummy.task.dummytaskstate import DummyTaskDefaults, DummyTaskOptions, DummyTaskDefinition, ls_R
+from apps.dummy.task.dummytaskstate import DummyTaskDefaults, DummyTaskOptions, DummyTaskDefinition
```
Comment (Wiezzel) [inline tests/apps/dummy/task/test_dummytaskstate.py]: Line too long.

File: tests/golem/resource/test_dirmanager.py [hunk 5]
```diff
@@ -275,3 +276,34 @@ def test_find_task_script(self):
...
+                         for x in [["ee", "f1"],
+                                   ["f2"],
+                                   ["aa", "bb", "f3"],
+                                   ["ee", "ff", "f4"]]}
+            self.assertEqual(set(dirs), true_dirs)
+        else:
+            dirs = list(list_dir_recursive(self.tempdir))
+            true_dirs = {os.path.join(*[self.tempdir, *x])
+                         for x in [["ee", "f1"], ["f2"], ["aa", "bb", "f3"]]}
+            self.assertEqual(set(dirs), true_dirs)
```
Comment (Wiezzel) [inline tests/golem/resource/test_dirmanager.py]: No newline at the end of file.

**Review by Wiezzel — CHANGES_REQUESTED** [review]
Please fix these linter errors.

File: golem/resource/dirmanager.py (hunk 1 above)
Comment (inexxt) [reply golem/resource/dirmanager.py]: Fixed

File: golem/resource/dirmanager.py (hunk 2 above)
Comment (inexxt) [reply golem/resource/dirmanager.py]: Fixed

File: golem/resource/dirmanager.py (hunk 3 above)
Comment (inexxt) [reply golem/resource/dirmanager.py]: Fixed

File: tests/apps/dummy/task/test_dummytaskstate.py (hunk 4 above)
Comment (inexxt) [reply tests/apps/dummy/task/test_dummytaskstate.py]: Fixed

File: tests/golem/resource/test_dirmanager.py (hunk 5 above)
Comment (inexxt) [reply tests/golem/resource/test_dirmanager.py]: Fixed

**Review by Wiezzel — APPROVED** [review]
OK

**Pull request merged**